rust/private/rustc.bzl

```
@@ -509,46 +505,46 @@ def rustc_compile_action(
             # nb. This field is required for cc_library to depend on our output.
             files = depset([crate_info.output]),
             runfiles = runfiles,
-            executable = crate_info.output if crate_info.type == "bin" or out_binary else None,
+            executable = crate_info.output if crate_info.type == "bin" or "--test" in rust_flags or out_binary else None,
         ),
     ]
 
 def add_edition_flags(args, crate):
     if crate.edition != "2015":
         args.add("--edition={}".format(crate.edition))
 
-def _create_out_dir_action(ctx, file, build_info, dep_info):
+def _create_extra_input_args(ctx, file, build_info, dep_info):
     tar_file_attr = getattr(file, "out_dir_tar", None)
```

```suggestion
    if getattr(file, "out_dir_tar", None):
      fail("`out_dir_tar` attribute is no longer supported, please rely on //cargo:cargo_build_script.bzl instead.")
```

Also you need to amend the rule definition to remove the doc of this attribute.